gui/wxpython/startup/guiutils.py

```diff
@@ -139,6 +148,107 @@ def _locationAlreadyExists(self, ctrl):
                caption=_("Existing location path"))
 
 
+def get_reasons_mapsets_not_removable(mapsets, check_permanent):
+    """Get reasons why mapsets cannot be removed.
+    Parameter *mapsets* is a list of tuples (database, location, mapset).
+    Parameter *check_permanent* is True of False. It depends on whether
+    we want to check for permanent mapset or not.
+
+    This is currently just a convenience wrapper for get_reasons_mapset_not_removable().
+    """
+    messages = []
+    for grassdb, location, mapset in mapsets:
+        messages += get_reasons_mapset_not_removable(grassdb, location,
+                                                     mapset, check_permanent)
+    return messages
+
+
+def get_reasons_mapset_not_removable(grassdb, location, mapset, check_permanent):
```

Comment:
I think we wanted "get_reason_mapset_not_removable", currently the implementation reports 1 reason because you have elif there and not if for every case. We could also report multiple reasons (e.g. mapset can be locked and PERMANENT), but I would keep single reason. Therefore the name should reflect that. It's a detail, but let's get it right.

Reply:
Also it should return message as string (or None), not list to keep it all consistent. Update the documentation.